### PR TITLE
refactor: change how `get_role` works

### PR DESCRIPTION
**Pull request by sillyElvy**
## About

This PR changes how `guild.get_role` works to prevent and eliminate rate limits

## Checklist

- [x] I've ran `pre-commit` to format and lint the change(s) made.
- [ ] I've checked to make sure the change(s) work on `3.8.6` and higher.
- [ ] This fixes/solves an [Issue](https://github.com/goverfl0w/discord-interactions/issues).
  - (If existent):
- [ ] I've made this pull request for/as: (check all that apply)
  - [ ] Documentation
  - [ ] Breaking change
  - [x] New feature/enhancement
  - [ ] Bugfix

### CODE REVIEW

**Comment by sillyElvy** [comment]
Yes, I became kinda addicted to `next()`

File: interactions/api/models/guild.py [hunk 1]
```diff
@@ -1638,14 +1638,21 @@ async def get_role(
...
+                raise LibraryException(code=13)
+            roles = await self._client.get_all_roles(guild_id=int(self.id))
+            try:
+                role = next(role for role in roles if int(role["id"]) == role_id)
+                self.roles = [Role(**_) for _ in roles]
+                return Role(**role)
+            except StopIteration as exc:
+                raise LibraryException(
+                    message="The role you looked for was not found!", code=0, severity=30
+                ) from exc
```
Comment (Toricane) [inline interactions/api/models/guild.py]:
```suggestion
        for role in self.roles:
            if int(role.id) == role_id:
                return role
        else:
            if not self._client:
                raise LibraryException(code=13)
            roles = await self._client.get_all_roles(guild_id=int(self.id))
            self.roles = [Role(**_) for _ in roles]
            for role in self.roles:
                if int(role.id) == role_id:
                    break
            else:
                raise LibraryException(
                    message="The role you looked for was not found!", code=0, severity=30
                )
            return role
```
Use for else
DISCLAIMER: did this on phone, pls test and format the code if necessary

**Review by Toricane — CHANGES_REQUESTED** [review]
Use for else
DISCLAIMER: did this on phone, pls test and format the code if necessary

File: interactions/api/models/guild.py (hunk 1 above)
Comment (sillyElvy) [reply interactions/api/models/guild.py]: What do you have against `next()`?

**Comment by Toricane** [comment]
Next keeps looping until the end; it's better to loop through and break early

File: interactions/api/models/guild.py (hunk 1 above)
Comment (Toricane) [reply interactions/api/models/guild.py]: Next loops till end of list
Breaking early in for loops is better
Also no catching exceptions, else is used instead

Comment (sillyElvy) [reply interactions/api/models/guild.py]: Is for/else even a thing?

Comment (Toricane) [reply interactions/api/models/guild.py]: Also the code is shorter

Comment (Toricane) [reply interactions/api/models/guild.py]: Yes, google it

Comment (sillyElvy) [reply interactions/api/models/guild.py]: Also please remove the `from exc` from the pre-last line

Comment (Toricane) [reply interactions/api/models/guild.py]: Else is run if you don't break from for loop

Comment (Toricane) [reply interactions/api/models/guild.py]: Ok

Comment (Toricane) [reply interactions/api/models/guild.py]: > Also please remove the `from exc` from the pre-last line

Done

Comment (sillyElvy) [reply interactions/api/models/guild.py]: Alright nice

**Comment by sillyElvy** [comment]
I'll revert the removal of next

**Comment by i0bs** [comment]
Why is the new module path in.. german?

**Comment by sillyElvy** [comment]
> Why is the new module path in.. german?

I accidentally pressed translate for Google chrome and it somehow changed everything while commiting - I'm about to fix it

**Comment by sillyElvy** [comment]
> > Why is the new module path in.. german?
> 
> I accidentally pressed translate for Google chrome and it somehow changed everything while commiting - I'm about to fix it 

Ok a chrome tab on mobile is not made for this lmao

**Comment by sillyElvy** [comment]
I fucked this branch too hard, making another rq

**Pull request closed without merging**